consensus/src/consensus/inner/agent.rs

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::consensus::ConsensusMessageWrapped;
+
+use super::*;
+
+impl ConsensusInner {
+    /// initialize genesis block if neccesary and catches up the chain
+    async fn init(&mut self) -> Result<()> {
+        let canon = self.storage.canon().await?;
+        // no blocks present/genesis situation
+        if canon.block_height == 0 && canon.hash.is_empty() {
```

Comment:
note: a new way of checking for the genesis block is in the works, might be a good idea to always use the same method for this: https://github.com/AleoHQ/snarkVM/pull/321